### ISSUE TITLE
Unable to save plugin settings in  2023.2 Beta

### ISSUE THREAD

**Issue by MCMicS** [MCMicS/jenkins-control-plugin#490]
IntelliJ IDEA 2023.2 Beta (Ultimate Edition)
Build #IU-232.8660.48, built on July 12, 2023
Runtime version: 17.0.7+7-b1000.6 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
Jenkins Plugin 0.13.18-eap3-2022.2

```
The plugin Jenkins Control Plugin failed to save settings. Please restart IntelliJ IDEA
java.lang.Exception: Cannot get JenkinsBrowserPanel component state at 
com.intellij.configurationStore.ComponentStoreImpl.commitComponents$suspendImpl(ComponentStoreImpl.kt:249) at 
com.intellij.configurationStore.ComponentStoreImpl$commitComponents$1.invokeSuspend(ComponentStoreImpl.kt) at 
kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) at 
kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) at 
kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584) at 
kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793) at 
kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697) at 
kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684) Caused by: com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: Access is allowed from Event Dispatch Thread (EDT) only; see https://jb.gg/ij-platform-threading for details Current thread: Thread[DefaultDispatcher-worker-1,5,main] 604688107 (EventQueue.isDispatchThread()=false) SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 1152848675 at 
com.intellij.openapi.application.impl.ApplicationImpl.createThreadAccessException(ApplicationImpl.java:1083) at 
com.intellij.openapi.application.impl.ApplicationImpl.throwThreadAccessException(ApplicationImpl.java:1078) at 
com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1066) at 
org.codinjutsu.tools.jenkins.JenkinsTree.getState(JenkinsTree.java:211) at 
org.codinjutsu.tools.jenkins.view.BrowserPanel.getState(BrowserPanel.java:495) at 
org.codinjutsu.tools.jenkins.view.BrowserPanel.getState(BrowserPanel.java:55) at 
com.intellij.configurationStore.ComponentStoreImpl$commitComponent$5.invoke(ComponentStoreImpl.kt:364) at 
com.intellij.openapi.application.rw.InternalReadAction.insideReadAction(InternalReadAction.kt:105) at 
com.intellij.openapi.application.rw.InternalReadAction.access$insideReadAction(InternalReadAction.kt:14) at 
com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$2.invoke(InternalReadAction.kt:95) at 
com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$2.invoke(InternalReadAction.kt:94) at 
com.intellij.openapi.application.rw.CancellableReadActionKt$cancellableReadActionInternal$1.invoke$lambda$1$lambda$0(cancellableReadAction.kt:41) at 
com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1133) at 
com.intellij.openapi.application.rw.CancellableReadActionKt$cancellableReadActionInternal$1.invoke$lambda$1(cancellableReadAction.kt:39) at 
com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:63) at 
com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:133) at 
com.intellij.openapi.application.rw.CancellableReadActionKt$cancellableReadActionInternal$1.invoke(cancellableReadAction.kt:37) at 
com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:248) at 
com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal(cancellableReadAction.kt:34) at 
com.intellij.openapi.application.rw.InternalReadAction.tryReadCancellable(InternalReadAction.kt:94) at 
com.intellij.openapi.application.rw.InternalReadAction.tryReadAction(InternalReadAction.kt:76) at 
com.intellij.openapi.application.rw.InternalReadAction.readLoop(InternalReadAction.kt:63) at 
com.intellij.openapi.application.rw.InternalReadAction.access$readLoop(InternalReadAction.kt:14) at 
com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invokeSuspend(InternalReadAction.kt:42) at 
com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invoke(InternalReadAction.kt) at 
com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invoke(InternalReadAction.kt) at 
kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78) at 
kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:167) at 
kotlinx.coroutines.BuildersKt.withContext(Unknown Source) at 
com.intellij.openapi.application.rw.InternalReadAction.runReadAction(InternalReadAction.kt:38) at 
com.intellij.openapi.application.rw.PlatformReadWriteActionSupport.executeReadAction(PlatformReadWriteActionSupport.kt:38) at 
com.intellij.openapi.application.ReadWriteActionSupport.executeReadAction$default(ReadWriteActionSupport.kt:15) at 
com.intellij.openapi.application.CoroutinesKt.constrainedReadAction(coroutines.kt:58) at 
com.intellij.openapi.application.CoroutinesKt.readAction(coroutines.kt:25) at 
com.intellij.configurationStore.ComponentStoreImpl.commitComponent(ComponentStoreImpl.kt:363) at 
com.intellij.configurationStore.ComponentStoreImpl.commitComponents$suspendImpl(ComponentStoreImpl.kt:245) ... 7 more
```

Steps:
1. Open Configuration
2. Click Test Connection
3. Click Apply Button

**Comment by MCMicS** [comment]
not happens with 2023.2 RC
IntelliJ IDEA 2023.2 RC (Ultimate Edition)
Build #IU-232.8660.142, built on July 21, 2023
Runtime version: 17.0.7+7-b1000.6 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
Jenkins Plugin 0.13.18-eap3-2022.2
GC: G1 Young Generation, G1 Old Generation

Kotlin: 232-1.9.0-IJ8660.142

**Comment by liamjones** [comment]
FYI they've just released 2023.2 fully and I'm seeing the same issue: https://blog.jetbrains.com/webstorm/2023/07/webstorm-2023-2/?utm_source=product&utm_medium=link&utm_campaign=TBA

**Comment by MCMicS** [comment]
can you please test https://plugins.jetbrains.com/plugin/6110-jenkins-control/versions/eap/367055

the error is gone with this build (I hope :smile:)

**Comment by liamjones** [comment]
@MCMicS Seems to be fixed, thanks! 😄 

(I did lose all my settings and had to reconfigure the plugin but that might just be normal with switching to an EAP - or because it couldn't save it lost stuff?)

**Comment by MCMicS** [comment]
Hi
normally no settings should be lost if settings was previously saved successfully

but nice to hear that it works now

**Comment by MCMicS** [comment]
released with [0.13.18](../releases/tag/v0.13.18)
